eq_translations/survey_schema.py

```diff
@@ -77,24 +77,37 @@ def _get_translatable_items_from_dict_element(
     ):
         plural_forms = schema_element.get("text_plural", {}).get("forms")
         if plural_forms:
-            plural_forms_pointer = f"{pointer}/text_plural/forms"
-            yield self._get_translatable_item(
-                plural_forms_pointer, plural_forms, with_context
+            yield from self._get_plural_translatable_item(
+                pointer, plural_forms, with_context
             )
-
-        for element, value in schema_element.items():
-            if isinstance(value, str):
+        elif "text" in schema_element:
+            yield from self._get_placeholder_text_translatable_item(
+                pointer, schema_element["text"], with_context
+            )
+        else:
+            for element, value in schema_element.items():
                 element_pointer = f"{pointer}/{element}"
                 yield self._get_translatable_item(element_pointer, value, with_context)
 
     def _get_translatable_items_from_list_element(
         self, pointer, schema_element, with_context
     ):
-        for index, value in enumerate(schema_element):
-            if isinstance(value, str):
+
+        for index, element in enumerate(schema_element):
+            if isinstance(element, dict):
+                plural_forms = element.get("text_plural", {}).get("forms")
+                if plural_forms:
+                    yield from self._get_plural_translatable_item(
+                        pointer, plural_forms, with_context
+                    )
+                elif "text" in element:
+                    yield from self._get_placeholder_text_translatable_item(
+                        pointer, element["text"], with_context
+                    )
```

Comment:
Could we not just call `_get_translatable_items_from_dict_element` if `element` is a dict?

Reply:
Changed 👍